### PR TITLE
III-3558 Replace datetime custom format

### PR DESCRIPTION
**Pull request by LucWollants**
### Changed
- Changed creating date times with custom format to wrapper method to avoid type conflict

---
Ticket: https://jira.uitdatabank.be/browse/III-3558

### CODE REVIEW

File: src/Event/Events/EventFromUDB2.php [hunk 1]
```diff
@@ -151,8 +152,8 @@ private function getCalendar(array $calendarAsArray): Calendar
                     $startTime;
             $subEvents[] = new SubEvent(
                 new DateRange(
-                    \DateTimeImmutable::createFromFormat('Y-m-d\TH:i:s', $startTime, new DateTimeZone('Europe/Brussels')),
-                    \DateTimeImmutable::createFromFormat('Y-m-d\TH:i:s', $endTime, new DateTimeZone('Europe/Brussels'))
+                    DateTimeFactory::fromFormat('Y-m-d\TH:i:s', $startTime, new DateTimeZone('Europe/Brussels')),
```
Comment (grubolsch) [inline src/Event/Events/EventFromUDB2.php:155]: Denk dat je hiervoor de core constant DATE_ISO8601 kan gebruiken. (https://www.php.net/manual/en/datetime.constants.php#constant.date-iso8601)

File: src/Event/Events/EventFromUDB2.php [hunk 2]
```diff
@@ -132,8 +133,8 @@ private function getCalendar(array $calendarAsArray): Calendar
 
         if ($calendarType === 'periods') {
             $dateRange = new DateRange(
-                \DateTimeImmutable::createFromFormat('Y-m-d', $calendarAsArray['periods'][0]['period'][0]['datefrom'][0]['_text']),
-                \DateTimeImmutable::createFromFormat('Y-m-d', $calendarAsArray['periods'][0]['period'][0]['dateto'][0]['_text'])
+                DateTimeFactory::fromFormat('Y-m-d', $calendarAsArray['periods'][0]['period'][0]['datefrom'][0]['_text']),
```
Comment (grubolsch) [inline src/Event/Events/EventFromUDB2.php:136]: Would it be a good idea to put this format in a constant?

File: src/Event/Events/EventFromUDB2.php [hunk 3]
```diff
@@ -132,8 +133,8 @@ private function getCalendar(array $calendarAsArray): Calendar
 
         if ($calendarType === 'periods') {
```
Comment (grubolsch) [inline src/Event/Events/EventFromUDB2.php:134]: Not code you changed now, but can we put a todo comment here that once we upgrade to PHP8 we can change calenderType to be an Enum?

**Review by simon-debruijn — APPROVED** [review]

**Review by grubolsch — APPROVED** [review]
Good for me, but I would add the formats as a constant, to prevent typo's/improve reusability.

File: src/Event/Events/EventFromUDB2.php (hunk 1 above)
Comment (LucWollants) [reply src/Event/Events/EventFromUDB2.php:155]: As discussed best to fix in dedicated ticket

File: src/Event/Events/EventFromUDB2.php (hunk 2 above)
Comment (LucWollants) [reply src/Event/Events/EventFromUDB2.php:136]: As discussed best to fix in dedicated ticket

**Pull request merged**